tests/testutils/README.md

```diff
@@ -0,0 +1,62 @@
+# Tests Utilities
+
+The `testutils` package provides an internal test format for Collector data, and helpers to help assert its integrity
+from arbitrary components.
+
+### Resource Metrics
+
+`ResourceMetrics` are at the core of the internal metric data format for these tests and are intended to be defined
+in yaml files or by converting from obtained `pdata.Metrics` items.
+
+```yaml
+resource_metrics:
+  - attributes:
+      a_resource_attribute: a_value
+      another_resource_attribute: another_value
+    ilms:
+      - instrumentation_library:
+          name: a library
+          version: some version
+      - metrics:
+          - name: my.int.gauge
+            type: IntGauge
+            description: my.int.gauge description
+            unit: ms
+            labels:
+              my_label: my_label_value
+              my_other_label: my_other_label_value
+            value: 123
```

Comment:
If I am testing a real third-party app (e.g. Postgres) do I really have a chance to know what values to expect exactly? Are values optional in test specification?

Reply:
You could expect particular values like number of instances or 1/0 flags denoting status.  My aim is to introduce `expr` expression support for expected/allowed ranges over time.